pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTaskRestletResource.java

```diff
@@ -231,6 +236,23 @@ public Map<String, PinotHelixTaskResourceManager.TaskDebugInfo> getTasksDebugInf
     return _pinotHelixTaskResourceManager.getTasksDebugInfoByTable(taskType, tableNameWithType, verbosity);
   }
 
+  @GET
+  @Path("/tasks/generator/{tableNameWithType}/{taskType}/debug")
+  @ApiOperation("Fetch task generation information for the recent runs of the given task for the given table")
+  public BaseTaskGeneratorInfo getTaskGenerationDebugInto(
+      @ApiParam(value = "Task type", required = true) @PathParam("taskType") String taskType,
+      @ApiParam(value = "Table name with type", required = true) @PathParam("tableNameWithType")
+          String tableNameWithType) {
+    BaseTaskGeneratorInfo
+        taskGeneratorMostRecentRunInfo = _taskManagerStatusCache.fetchTaskGeneratorInfo(tableNameWithType, taskType);
```

Comment:
What if the table is not mastered on this controller?

Reply:
> @saurabhd336 thanks a lot for your contribution. It has been hard to debug minions, so I totally understand your pain point. However, I have a few general questions:
> 
> 1. What is the use case where you think this will help? (we have some use cases where we saw problems, and instead of introducing one new API per use case, I am looking to make sure we do something that works for more than one. @sajjad-moradi  can review from our standpoint).
> 2. I know @npawar referred to a debug API on the minion, but I could not locate one. Please correct me if I am wrong. I think we should issue an API to the minions as well (liek we do with servers) and get the latest status form them and display it in the controller.

Hey Subbu, this is the one for Minion tasks: `/tasks/task/{taskName}/debug`. In the list of subtasks returned, there's an "info" field which contains the exception (if any) encountered by this task

Reply:
ah good catch about the lead controller! this won't work as is, we'll have to think of how to handle that

Reply:
for 2. I feel it's not very convenient when one auto scales minion workers frequently based on pending tasks. The debug API would be either not reachable when no workers are around or get no past task states to check when new workers are added

Reply:
But we have helix APIs for that, right? Has that avenue been fully explored? We are migrating to Helix 1.x) very soon (we are getting to production with that), so please check in 1.x pages what is available for task management and debugging, thanks. We will also do the same.
cc: @sajjad-moradi

Reply:
There are two parts:
- generating task configurations which happens in pinot code
- submitting the task via TaskDriver which is in Helix code
Subbu has a valid point that helix should provide some info on failure on the 2nd part - task submission. The first part though is before reaching helix and it's good if we have an API to capture issues in task generation in pinot code.

Reply:
Also scheduling tasks can be done by either:
- periodic task generator job on lead controller, or
- task REST endpoint which can be on any controller

So this means the approach in this PR doesn't work as expected.